pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -415,6 +437,120 @@ def newpkgcb(pkgId, name, arch):
                     dc = DeclarativeContent(content=default_content, d_artifacts=[da])
                     await self.put(dc)
 
+            if comps_downloader:
+                comps_result = await comps_downloader.run()
+
+                comps = libcomps.Comps()
+                comps.fromxml_f(comps_result.path)
+
+                comps_pb.total = (
+                    len(comps.groups) + len(comps.categories) + len(comps.environments)
+                )
+                comps_pb.state = 'running'
+                comps_pb.save()
+
+                if comps.langpacks:
+                    langpack_dict = PackageLangpacks.libcomps_to_dict(comps.langpacks)
+                    prep_hash = list(langpack_dict.values())
+                    str_prep_hash = [str(i) for i in prep_hash]
+                    str_prep_hash.sort()
+                    langpack_digest = hashlib.sha256(
+                        ''.join(str_prep_hash).encode('utf-8')
+                    ).hexdigest()
```

Comment:
This also looks like a candidate for a helper function, since you do the same stuff multiple times.